src/toil/worker.py

```diff
@@ -259,10 +259,6 @@ def badWorker():
                 else:
                     jobGraph.command = jobGraph.checkpoint
 
-                # Reduce the retry count
-                assert jobGraph.remainingRetryCount >= 0
-                jobGraph.remainingRetryCount = max(0, jobGraph.remainingRetryCount - 1)
```

Comment:
I'm a bit confused, how did remainingRetryCount every get below 0? The advantage of doing this here is there is less I/O on the leader, and yes, its a bit squishy with the max statement